worker.go

```diff
@@ -303,21 +303,18 @@ func (w *Worker) Stop() {
 
 // EventWorker is similar to a Worker but it collects events and service checks instead of metrics.
 type EventWorker struct {
-	EventChan        chan samplers.UDPEvent
-	ServiceCheckChan chan samplers.UDPServiceCheck
-	mutex            *sync.Mutex
-	events           []samplers.UDPEvent
-	checks           []samplers.UDPServiceCheck
-	traceClient      *trace.Client
+	SampleChan  chan ssf.SSFSample
```

Comment:
I kiiinda want this to be internal only and take a chan we pass in, much like we do with the new SpanWorker... Parallelizing it with that in place was much easier (also feels "right-er" to me?)